docs/docs/concepts/tool_calling.mdx

```diff
@@ -0,0 +1,133 @@
+# Tool Calling
```

Comment:
We need a URL that we can link to that explains what a Tool is / brief example of definition/ brief example of how it can be used.

I think we need another page for tools specifically, or else retitle this page to `# Tools and have a tool calling section somewhere within it?

Some of the questions that the conceptual guide should address:

1) What is a tool?
2) Where is the tool getting executed? (maybe does a tool code need to be executed?)
3) How does one define a tool? (We have 4-5 ways of doing it); e.g., @tool  decorator, StructuredTool.from_function..... We don't need to do it in detail, but can link to the how-to guide, but just explain high level what the differences are.

Reply:
that's true. let me try to add as a section on this page.

Reply:
also noted in Slack discussion. this is a good thing to flesh out -- 
https://langchain.slack.com/archives/C04GWPE38LV/p1728407895162059